com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateServiceTagsStep.java

```diff
@@ -0,0 +1,72 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceInstanceGetter;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceUpdater;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+public class UpdateServiceTagsStep extends AsyncFlowableStep {
+
+    @Inject
+    @Named("serviceUpdater")
+    protected ServiceUpdater serviceUpdater;
+
+    @Inject
+    private ServiceInstanceGetter serviceInstanceGetter;
+
+    @Override
+    protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exception {
+        CloudServiceExtended serviceToUpdate = StepsUtil.getServiceToProcess(execution.getContext());
+        ResponseEntity<String> updateServiceResponse = updateServiceTags(execution.getControllerClient(), serviceToUpdate);
+        
+        if(HttpStatus.ACCEPTED.equals(updateServiceResponse.getStatusCode())) {
+            Map<String, ServiceOperationType> serviceOperation = new HashMap<>();
+            serviceOperation.put(serviceToUpdate.getName(), ServiceOperationType.UPDATE);
+            
+            execution.getStepLogger()
+            .debug(Messages.TRIGGERED_SERVICE_OPERATIONS, JsonUtil.toJson(serviceOperation, true));
+            StepsUtil.setTriggeredServiceOperations(execution.getContext(), serviceOperation);
+            
+            return StepPhase.POLL;
+        }
+        return StepPhase.DONE;
+    }
+
+    private ResponseEntity<String> updateServiceTags(CloudControllerClient client, CloudServiceExtended service) {
+        // TODO: Remove the service.isUserProvided() check when user provided services support tags.
+        // See the following issue for more info:
+        // https://www.pivotaltracker.com/n/projects/966314/stories/105674948
+        if (service.isUserProvided()) {
+            return new ResponseEntity<String>(HttpStatus.CREATED);
```

Comment:
What?

Reply:
Done and switched to MethodExecution response.